spec/strategies/header_strategy_spec.rb

```diff
@@ -2,17 +2,19 @@
 
 describe Phenotype::HeaderStrategy do
   describe "#get_version" do
-    context "with the right header" do
+    context "with dynamic header" do
       let(:env) do
         env = Rack::MockRequest.env_for('/foo')
-        env['HTTP_SUE'] = 'v1'
+        env['HTTP_VERSION'] = 'v1'
         env
       end
 
-      subject { described_class.new(header: 'Sue') }
+      subject { described_class.new(header: 'version') }
 
-      it "returns with v1" do
+      it "returns correct version" do
         expect(subject.get_version(env)).to eql("v1")
+        env['HTTP_VERSION'] = 'v2'
+        expect(subject.get_version(env)).to eql("v2")
```

Comment:
these tests must be run together so subject instance is cached (mimics config.ru which only creates on instance of the header strategy)